girder/models/model_base.py

```diff
@@ -736,7 +750,96 @@ def setPublic(self, doc, public, save=False):
 
         return doc
 
-    def setAccessList(self, doc, access, save=False):
+    def setPublicFlags(self, doc, flags, user, append=False, save=False, force=False):
+        """
+        Set permission flags that are granted on this resource to anonymous users.
+        This means any user, whether anonymous or logged in, will receive all
+        of the specified permissions. This also validates that the user attempting
+        to set the flags has permission to do so.
+
+        :param doc: The document to update permission flags on.
+        :type doc: dict
+        :param flags: Flags or set of flags to add.
+        :type flags: flag identifier, or a list/set/tuple of them
+        :param user: The user performing this action.
+        :type user:
+        :param append: Whether to append to the list or replace it.
+        :type append: bool
+        :param save: Whether to save the document to the database afterward.
+        :type save: bool
+        :param force: Set this to True to disable validation of the current
```

Comment:
I find this explanation confusing.

How about

`Set this to True if you explicitly want to circumvent validation of the passed in user's ability to set the flags.`

or even simpler

`Set this to True to set the flags regardless of the passed in user's permissions.`